clarity/src/vm/functions/database.rs

```diff
@@ -924,3 +947,188 @@ pub fn special_get_burn_block_info(
         }
     }
 }
+
+/// Handles the `get-stacks-block-info?` special function.
+/// Interprets `args` as variables `[property-name, block-height]`, and returns
+/// a property value determined by `property-name`:
+/// - `id-header-hash` returns the index block hash at `block-height`
+/// - `header-hash` returns the header hash at `block-height`
+/// - `time` returns the block time at `block-height`
+///
+/// # Errors:
+/// - CheckErrors::IncorrectArgumentCount if there aren't 2 arguments.
+/// - CheckErrors::GetStacksBlockInfoExpectPropertyName if `args[0]` isn't a ClarityName.
+/// - CheckErrors::NoSuchStacksBlockInfoProperty if `args[0]` isn't a StacksBlockInfoProperty.
+/// - CheckErrors::TypeValueError if `args[1]` isn't a `uint`.
+pub fn special_get_stacks_block_info(
+    args: &[SymbolicExpression],
+    env: &mut Environment,
+    context: &LocalContext,
+) -> Result<Value> {
+    // (get-stacks-block-info? property-name block-height-uint)
+    runtime_cost(ClarityCostFunction::BlockInfo, env, 0)?;
+
+    check_argument_count(2, args)?;
+
+    // Handle the block property name input arg.
+    let property_name = args[0]
+        .match_atom()
+        .ok_or(CheckErrors::GetStacksBlockInfoExpectPropertyName)?;
+
+    let block_info_prop = StacksBlockInfoProperty::lookup_by_name(property_name).ok_or(
+        CheckErrors::NoSuchStacksBlockInfoProperty(property_name.to_string()),
+    )?;
+
+    // Handle the block-height input arg.
+    let height_eval = eval(&args[1], env, context)?;
+    let height_value = match height_eval {
+        Value::UInt(result) => Ok(result),
+        x => Err(CheckErrors::TypeValueError(TypeSignature::UIntType, x)),
+    }?;
+
+    let Ok(height_value) = u32::try_from(height_value) else {
+        return Ok(Value::none());
+    };
+
+    let current_block_height = env.global_context.database.get_current_block_height();
+    if height_value >= current_block_height {
+        return Ok(Value::none());
+    }
+
+    let result = match block_info_prop {
+        StacksBlockInfoProperty::Time => {
+            let block_time = env.global_context.database.get_block_time(height_value)?;
+            Value::UInt(u128::from(block_time))
+        }
+        StacksBlockInfoProperty::HeaderHash => {
+            let header_hash = env
+                .global_context
+                .database
+                .get_block_header_hash(height_value)?;
+            Value::Sequence(SequenceData::Buffer(BuffData {
+                data: header_hash.as_bytes().to_vec(),
+            }))
+        }
+        StacksBlockInfoProperty::IdentityHeaderHash => {
```

Comment:
`id-block-hash` is actually short for "index block hash", not "identity block hash"

Reply:
Let's rename it.

Reply:
Done in e496acd2e4068216881c824dc087dab83dee632c.